lib/pbench/server/api/resources/query_controllers.py

```diff
@@ -0,0 +1,273 @@
+        """
+        _genMonthUriRange Construct a comma-separated list of index names
+        qualified by year and month suitable for use in the Elasticsearch
+        /_search query URI.
+
+        The month is incremented by 1 from "start" to "end"; for example,
+
+        _genMonthUriRange('drb.', 'v4.run.', '2020-08', '2020-10') will result
+        in
+
+            'drb.v4.run.2020-08,drb.v4.run.2020-09,drb.v4.run.2020-10,'
+
+        This code is adapted from Javascript in the Pbench dashboard's
+        'moment_constants.js' getAllMonthsWithinRange function.
+
+        Args:
+            prefix ([string]): The Pbench server index prefix
+            index ([string]): The desired monthly index root
+            start ([datetime.datetime]): The start time
+            end ([datetime.datetime]): The end time
+
+        Returns:
+            [string]: A comma-separated list of month-qualified index names
+        """
+        monthResults = list()
+        queryString = ""
+        for m in rrule.rrule(rrule.MONTHLY, dtstart=start, until=end):
+            monthResults.append(m.strftime("%Y-%m"))
+        for monthValue in monthResults:
+            if index == "v4.result-data.":
+                queryString += f"{prefix + index + monthValue}-*,"
+            else:
+                queryString += f"{prefix + index + monthValue},"
+        return queryString
+
+    def post(self):
+        """
+        POST for Pbench controller datasets with start times within the
+        specified search range, and owned by a specified user:
+
+        {
+            "user": "email-tag",
+            "start": "start-time",
+            "end": "end-time"
+        }
+
+        "user" specifies the owner of the data to be searched; it need not
+        necessarily be the user represented by the session token header,
+        assuming the session user is authorized to view "user"s data. This
+        will be used to constrain the Elasticsearch query.
+
+        TODO: When we have authorization infrastructure, we'll need to
+        check that "session user" has rights to view "user" data.
+
+        "start" is a time string representing the starting range of dates
+        to search.
+
+        "end" is a time string representing the end range of dates to
+        to search.
+
+        The Elasticsearch URL string is also dependent upon the configured
+        "prefix". This is a Pbench artifact that allows multiple Pbench
+        instances to share a single Elasticsearch server, by segregating
+        their data into unique namespaces. Each Pbench index name is
+        qualified by the index_prefix of the Pbench server that originally
+        indexed the data. The prefix comes from the pbench-server.cfg
+        file for the server instance. For example, the official staging
+        server index_prefix is "staging-pbench" so that data is separate
+        from the production server even though they share a single instance
+        of Elasticsearch.
```

Comment:
While the sentiment of this last sentence is correct, the specifics are not quite as described.  It might better to document this hypotheically, ... "For example. a staging server `index_prefix` might be `staging-pbench`, while a production server `index_prefix` might be `prod-pbench`, so that data from the two pbench-server instances does not conflict in the same Elasticsearch instance."   ... Or something like that.